server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorBalancer.java

```diff
@@ -128,16 +128,25 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
 
           if (holder != null) {
             moveSegment(segmentToMove, holder.getServer(), params);
+          } else {
+            ++unmoved;
           }
         }
       }
+      if(unmoved == maxSegmentsToMove) {
+        // Cluster should be alive and constantly adjusting
+        log.warn("No good moves found in tier [%s]", tier);
```

Comment:
This will probably do a lot of spurious warnings in really small clusters (where it's impossible/unlikely to find good moves). Could this be made a debug? Or only log for clusters above a certain size?